app/models/product.rb

```diff
@@ -0,0 +1,8 @@
+# Documentation for the Product class.
+class Product < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_and_belongs_to_many :categories
```

Comment:
- [ ] I had to re-read the docs and I found that both will actually work. However, I also found that the ```has_and_belongs_to_many``` bests fits here based on the fact that there will be a many to many relationships being established. This is also captured in the provided ERD when I rechecked. I, therefore, agree with you that the ```has_and_belongs_to_many``` is the best association for this scenario.

![image](https://github.com/Caren-Koroeny/BudgetApp/assets/86318284/92d958d9-0bb0-4cc8-a6f4-daf006aebe70)